testsuite/pytests/sli2py_regressions/test_issue_521.py

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+#
+# test_issue_521.py
+#
+# This file is part of NEST.
+#
+# Copyright (C) 2004 The NEST Initiative
+#
+# NEST is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# NEST is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Regression test for Issue #521 (GitHub).
+"""
+
+import pytest
+
+import nest
+
+
+@pytest.mark.skipif_missing_threads
+@pytest.mark.parametrize("num_threads", [1, 2, 4, 8, 16])
+def test_hpc_synapse_correct_global_target_node_id_multithreaded(num_threads):
+    """
+    Ensure that *_hpc connections also return correct global target node ID on ``GetConnections``.
+    """
+
+    nest.ResetKernel()
+    nest.local_num_threads = num_threads
+
+    nc = nest.Create("iaf_psc_alpha", 20)
+
+    # connect two neurons (prime number node ids to increase variations of thread placement)
```

Comment:
The comment is confusing, since 4 and 12 are not prime numbers (for once, zero indexing really gets confusing here). I think it would actually be better to just choose 4 and 12 here as is, and then use prime numbers of threads, e.g., 1, 2, 5, 11 (I don't think we need more).

Reply:
Fixed in [2d7a609](https://github.com/nest/nest-simulator/pull/2922/commits/2d7a609c8e88f4d08211d38a864271236d878857).